crates/bevy_transform/src/hierarchy/hierarchy.rs

```diff
@@ -44,17 +44,27 @@ impl Command for DespawnRecursive {
 
 pub trait DespawnRecursiveExt {
     /// Despawns the provided entity and its children.
-    fn despawn_recursive(&mut self);
+    fn despawn_recursive(self);
 }
 
 impl<'w, 's, 'a> DespawnRecursiveExt for EntityCommands<'w, 's, 'a> {
     /// Despawns the provided entity and its children.
-    fn despawn_recursive(&mut self) {
+    fn despawn_recursive(mut self) {
         let entity = self.id();
         self.commands().add(DespawnRecursive { entity });
     }
 }
 
+impl<'w> DespawnRecursiveExt for EntityMut<'w> {
+    /// Despawns the provided entity and its children.
+    fn despawn_recursive(mut self) {
+        let entity = self.id();
+        unsafe {
```

Comment:
This should have a `// SAFE: rationale` comment above it explaining why this is safe in relationship to the requirements of `self.world_mut()` (ex: EntityMut is consumed so even though the location is no longer valid, it cannot be accessed again with the invalid location).

Reply:
I'll just copy that rationale in.